source/http_stream.c

```diff
@@ -14,15 +14,437 @@
+            napi_get_value_string_utf8(env, node_value, (char *)value_buf.buffer, value_buf.capacity, &value_buf.len)) {
+            napi_throw_error(env, NULL, "HTTP header could not be extracted");
+            goto argument_error;
+        }
+        struct aws_http_header header = {.name = aws_byte_cursor_from_buf(&name_buf),
+                                         .value = aws_byte_cursor_from_buf(&value_buf)};
+        aws_http_message_add_header(request, header);
+    }
+    aws_byte_buf_clean_up(&name_buf);
+    aws_byte_buf_clean_up(&value_buf);
+
+    struct aws_napi_callback on_complete;
+    AWS_ZERO_STRUCT(on_complete);
+    if (aws_napi_callback_init(&on_complete, env, node_args[5], "aws_http_stream_on_complete", s_on_complete_params)) {
+        napi_throw_error(env, NULL, "on_complete must be a callback");
+        return NULL;
+    }
+
+    struct aws_napi_callback on_response;
+    AWS_ZERO_STRUCT(on_response);
+    if (!aws_napi_is_null_or_undefined(env, node_args[6])) {
+        if (aws_napi_callback_init(
+                &on_response, env, node_args[6], "aws_http_stream_on_response", s_on_response_params)) {
+            napi_throw_error(env, NULL, "Unable to bind on_response callback");
+            return NULL;
+        }
+    }
+
+    struct aws_napi_callback on_body;
+    AWS_ZERO_STRUCT(on_body);
+    if (!aws_napi_is_null_or_undefined(env, node_args[7])) {
+        if (aws_napi_callback_init(&on_body, env, node_args[7], "aws_http_stream_on_body", s_on_body_params)) {
+            napi_throw_error(env, NULL, "Unable to bind on_body callback");
+            return NULL;
+        }
+    }
+
+    struct http_stream_binding *binding = aws_mem_calloc(allocator, 1, sizeof(struct http_stream_binding));
+    if (!binding) {
+        aws_napi_throw_last_error(env);
+        goto failed_binding_alloc;
+    }
+
+    binding->body = body;
+    binding->body_stream = body_stream;
+    binding->on_complete = on_complete;
+    binding->on_response = on_response;
+    binding->on_body = on_body;
+    binding->request = request;
+
+    struct aws_http_make_request_options request_options = {.self_size = sizeof(struct aws_http_make_request_options),
+                                                            .request = request,
+                                                            .user_data = binding,
+                                                            .on_response_headers = s_on_response_headers,
+                                                            .on_response_header_block_done =
+                                                                s_on_response_header_block_done,
+                                                            .on_response_body = s_on_response_body,
+                                                            .on_complete = s_on_complete,
+                                                            .manual_window_management = false};
```

Comment:
Toss a trailing comma on this bad boi and the formatter won't murder it so hard

Reply:
oh shit good idea.